### PR TITLE
Added section & environment folding to the latex parser

### PR DESCRIPTION
**Pull request by Nidish96**
I've added section (& subsection, ...) and environment (generic_environment & math_environment) folding to the latex parser.

### CODE REVIEW

File: ts-fold-parsers.el [hunk 1]
```diff
@@ -359,7 +359,12 @@
 
 (defun ts-fold-parsers-latex ()
   "Rule set for LaTex."
-  '((curly_group . ts-fold-range-seq)
+  '((generic_environment . ts-fold-range-latex-environment)
+    (math_environment . ts-fold-range-latex-environment)
+    (section		 . ts-fold-range-latex-section)
+    (subsection		 . ts-fold-range-latex-section)
+    (subsubsection	 . ts-fold-range-latex-section)
+    (curly_group	 . ts-fold-range-seq)
```
Comment (jcs090218) [inline ts-fold-parsers.el:367]: I think you are using tabs for some places. Can you make sure to use spaces instead?

File: ts-fold-parsers.el [hunk 2]
```diff
@@ -501,6 +506,19 @@
...
+  '((expression_list     . ts-fold-range-seq)
+    (function_definition . ts-fold-range-matlab-function)
+    (class_definition	 . ts-fold-range-matlab-function)
+    (if_statement	 . ts-fold-range-matlab-statements)
+    (for_statement	 . ts-fold-range-matlab-statements)
+    (while_statement	 . ts-fold-range-matlab-statements)
+    (switch_statement	 . ts-fold-range-matlab-statements)
+    (try_statement	 . ts-fold-range-matlab-statements)
+    (comment		 . ts-fold-range-matlab-blocks)
+    ))
```
Comment (jcs090218) [inline ts-fold-parsers.el:520]: Please resolve lint warnings:

```
Closing parens should not be wrapped onto new lines.
```

**Comment by jcs090218** [comment]
Can you forward declare functions at the very top. See

https://github.com/emacs-tree-sitter/ts-fold/blob/59c57057ba892e169510052d59921b8b3090fce9/ts-fold-parsers.el#L38-L97

Make sure there is no compile error:

```
ts-fold-parsers.el:509:8: Warning: function `ts-fold-parsers-matlab' defined multiple times in this file
```

You can [Eask](https://github.com/emacs-eask/cli) to do the clean compilation. See the [development](https://github.com/emacs-tree-sitter/ts-fold?tab=readme-ov-file#-development) section for more information.

File: ts-fold-parsers.el [hunk 3]
```diff
@@ -501,6 +508,18 @@
...
+  "Rule set for MATLAB."
+  '((expression_list     . ts-fold-range-seq)
+    (function_definition . ts-fold-range-matlab-function)
+    (class_definition	 . ts-fold-range-matlab-function)
+    (if_statement	 . ts-fold-range-matlab-statements)
+    (for_statement	 . ts-fold-range-matlab-statements)
+    (while_statement	 . ts-fold-range-matlab-statements)
+    (switch_statement	 . ts-fold-range-matlab-statements)
+    (try_statement	 . ts-fold-range-matlab-statements)
+    (comment		 . ts-fold-range-matlab-blocks)))
```
Comment (jcs090218) [inline ts-fold-parsers.el:521]: I've moved this function to the top (that's why there is the warning `Warning: function `ts-fold-parsers-matlab' defined multiple times in this file`). Make sure these functions are in alphabetic order!

Other than that, there are tabs in this function! 🤔

File: ts-fold.el [hunk 4]
```diff
@@ -1154,6 +1154,27 @@ more information."
...
+(defun ts-fold-range-latex-environment (node offset)
+  "Define fold range for latex environments.
+
+For arguments NODE and OFFSET, see function `ts-fold-range-seq' for
+more information."
+  (when-let* ((beg-node (tsc-get-child-by-field node :begin))
+              (end-node (tsc-get-child-by-field node :end))
+              (beg (tsc-node-end-position beg-node))
+              (end (tsc-node-start-position end-node)))
+    (ts-fold--cons-add (cons beg end) offset)))
```
Comment (jcs090218) [inline ts-fold.el:1176]: Can you place these functions in alphabetic order? Thank you!

**Comment by Nidish96** [comment]
Apologies - done now.

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
Thank you!

**Pull request merged**